labman/gui/handlers/process_handlers/quantification_process.py

```diff
@@ -27,46 +27,36 @@ def get(self):
 
     @authenticated
     def post(self):
-        plate_id = self.get_argument('plate-select')
-        # Magic number 0 -> there is only 1 file attached
-        file_content = self.request.files[
-            'plate-reader-fp'][0]['body'].decode('utf-8')
-        concentrations = QuantificationProcess.parse(file_content)
-        plate = Plate(plate_id)
-        self.render('quantification.html',
-                    concentrations=concentrations.tolist(),
-                    plate_name=plate.external_id, plate_id=plate.id,
-                    plate_conf=plate.plate_configuration.description)
+        # We will receive as many files as plates the user has selected
+        # The key of the self.request.files dictionary is of the form
+        # plate-file-<PLATE_ID> so use the keys to know the plates
+        # that we need to quantify
+        plates = []
+        for key in self.request.files:
+            plate_id = key.rsplit('-', 1)[1]
+            # The 0 is because for each key we have a single file
+            file_content = self.request.files[key][0]['body'].decode('utf-8')
+            plate = Plate(plate_id)
+            pc = plate.plate_configuration
+            concentrations = QuantificationProcess.parse(
```

Comment:
what happens if one of the files is malformed, should this be wrapped in a try/except? would a rollback be necessary?

Reply:
There is no modification of the DB in this section of the code (i.e. no rollback necessary).
If the file is malformed an error is raised, then tornado handles it and it is shown to the user in the bootrstapAlert